src/overlays/actors/ovl_En_Ruppecrow/z_en_ruppecrow.c

```diff
@@ -81,62 +112,558 @@ static DamageTable D_80BE39EC = {
+            this->rupees[rupeeIndex] = rupee;
+            this->rupees[rupeeIndex]->actor.gravity = -5.0f;
+            this->rupees[rupeeIndex]->actor.velocity.y = 0.0f;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_RUPY_FALL);
+            rupee = this->rupees[rupeeIndex];
+            rupee->unk152 = 60;
+            this->rupees[rupeeIndex]->actor.flags |= 0x10;
+        } else {
+            rupee = (EnItem00*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00,
+                                           this->actor.world.pos.x + xOffset, this->actor.world.pos.y,
+                                           this->actor.world.pos.z, 0x0, 0x0, 0x0, ITEM00_RUPEE_BLUE);
+            this->rupees[rupeeIndex] = rupee;
+            this->rupees[rupeeIndex]->actor.gravity = -5.0f;
+            this->rupees[rupeeIndex]->actor.velocity.y = 0.0f;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_RUPY_FALL);
+            rupee = this->rupees[rupeeIndex];
+            rupee->unk152 = 60;
+            this->rupees[rupeeIndex]->actor.flags |= 0x10;
+        }
+    } else if (this->rupeeIndex == 19) {
+        rupee =
+            (EnItem00*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00, this->actor.world.pos.x + xOffset,
+                                   this->actor.world.pos.y, this->actor.world.pos.z, 0x0, 0x0, 0x0, ITEM00_RUPEE_RED);
+        this->rupees[rupeeIndex] = rupee;
+        this->rupees[rupeeIndex]->actor.gravity = -5.0f;
+        this->rupees[rupeeIndex]->actor.velocity.y = 0.0f;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_RUPY_FALL);
+        rupee = this->rupees[rupeeIndex];
+        rupee->unk152 = 60;
+        this->rupees[rupeeIndex]->actor.flags |= 0x10;
+    } else {
+        rupee =
+            (EnItem00*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00, this->actor.world.pos.x + xOffset,
+                                   this->actor.world.pos.y, this->actor.world.pos.z, 0x0, 0x0, 0x0, ITEM00_RUPEE_GREEN);
+        this->rupees[rupeeIndex] = rupee;
+        this->rupees[rupeeIndex]->actor.gravity = -5.0f;
+        this->rupees[rupeeIndex]->actor.velocity.y = 0.0f;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_RUPY_FALL);
+        rupee = this->rupees[rupeeIndex];
+        rupee->unk152 = 60;
+        this->rupees[rupeeIndex]->actor.flags |= 0x10;
+    }
+
+    this->rupeeIndex++;
+}
+
+void EnRuppecrow_UpdatePosition(EnRuppecrow* this, GlobalContext* globalCtx) {
+    Vec3s nextPointDirection;
+
+    EnRuppecrow_GetPointDirection(this->path, this->currentPoint, &this->actor.world, &nextPointDirection);
+    if (this->actor.bgCheckFlags & 0x8) {
+        nextPointDirection.y = this->actor.wallYaw;
+    }
+
+    Math_SmoothStepToS(&this->actor.world.rot.y, nextPointDirection.y, 0x4, 0x3E8, 0x1);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    Math_SmoothStepToS(&this->actor.world.rot.x, -nextPointDirection.x, 0x4, 0x3E8, 0x1);
```

Comment:
We usually use dec if it is a round number
```suggestion
    Math_SmoothStepToS(&this->actor.world.rot.y, nextPointDirection.y, 4, 1000, 1);
    this->actor.shape.rot.y = this->actor.world.rot.y;
    Math_SmoothStepToS(&this->actor.world.rot.x, -nextPointDirection.x, 4, 1000, 1);
```

Reply:
But these are angles, so hex, right?